crates/bevy_pbr/src/material.rs

```diff
@@ -312,6 +321,33 @@ impl<M: Material, const I: usize> EntityRenderCommand for SetMaterialBindGroup<M
     }
 }
 
+pub fn verify_material_mesh_attributes<M: Material>(
+    material_meshes: Query<
+        &Handle<Mesh>,
+        (
+            With<Handle<M>>,
+            Or<(Changed<Handle<M>>, Changed<Handle<Mesh>>)>,
+        ),
+    >,
```

Comment:
I don't remember if `Changed<Handle<T>>` will detect only when the `Handle<T>` component itself is modified, i.e. if the handle component is overwritten to point at a different asset, or if it would also be matched when the asset referenced by the `Handle<T>` is mutated. I feel like it would only match the former and not the latter, as there are `AssetEvents`, and so I think it is missing some cases where Mesh or Material assets themselves are modified.